airbyte-cdk/python/airbyte_cdk/sources/declarative/extractors/csv_extractor.py

```diff
@@ -0,0 +1,53 @@
+#
+# Copyright (c) 2023 Airbyte, Inc., all rights reserved.
+#
+
+import csv
+from dataclasses import dataclass
+from io import StringIO
+from typing import Any, Iterable, Mapping
+
+import requests
+from airbyte_cdk.sources.declarative.extractors.record_extractor import RecordExtractor
+
+
+@dataclass
+class CSVExtractor(RecordExtractor):
+    """
+    Record extractor that takes a response in, assumes it's in CSV format, and returns the array of records.
+
+    Examples:
+    ```
+      extractor:
+        type: CSVExtractor
+    ```
+
+    ```
+      extractor:
+        type: Extractor
+        separator: ","
+        quotechar: '"'
+    ```
+
+    Attributes:
+        separator (str): Record separator in the incoming CSV.
+        quotechar (str): Quotation character used in the CSV.
+    """
+
+    separator: str
+    quotechar: str
+
+    def extract_records(self, response: requests.Response) -> Iterable[Mapping[str, Any]]:
+        try:
+            csv_data = StringIO(response.text)
```

Comment:
Despite the fact you've created the buffer here, [response body is still downloaded in full](https://requests.readthedocs.io/en/latest/user/advanced/#body-content-workflow) in httpclient (it is default behaviour, unless `stream=True` s is passed).